### PR TITLE
Metadata edgecase

### LINKED ISSUE #827: biom convert: "could not convert string to float" when first element in a metadata variable is blank
Hi

I get an error `ValueError: could not convert string to float: 'k__test;p__test'` when trying to convert from a delimited text OTU table to JSON. The error occurs when the first element in a metadata variable is blank, no matter what the rest contains. If the first element is a string, however, it's all merry christmas. 

The exact command I used was `biom convert -i otutable.txt -o output.biom --table-type="OTU table" --to-json --process-obs-metadata taxonomy`. Below are two minimal example files where the first works as intented, while the second results in the attached error:

## no error
```
#OTU ID	s1	s2	taxonomy
1	123	32	k__test;p__test
2	315	3	
3	0	22	k__test;p__test
```

## error
```
#OTU ID	s1	s2	taxonomy
1	123	32	
2	315	3	k__test;p__test
3	0	22	k__test;p__test
```

<details><summary>traceback (stderr output)</summary>
<p>

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4775, in _extract_data_from_tsv
    values = list(map(dtype, fields[1:]))
ValueError: could not convert string to float: 'k__test;p__test'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 660, in load_table
    table = parse_biom_table(fp)
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 412, in parse_biom_table
    t = Table.from_tsv(fp, None, None, lambda x: x)
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4631, in from_tsv
    t_md_name) = Table._extract_data_from_tsv(lines, **kwargs)
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4779, in _extract_data_from_tsv
    raise TypeError(msg % (lineno, badidx+1, badval))
TypeError: Invalid value on line 2, column 3, value k__test;p__test

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/biom", line 11, in <module>
    load_entry_point('biom-format==2.1.7', 'console_scripts', 'biom')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/biom/cli/table_converter.py", line 114, in convert
    table = load_table(input_fp)
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 662, in load_table
    raise TypeError("%s does not appear to be a BIOM file!" % f)
TypeError: otutable_wblankfirst.txt does not appear to be a BIOM file!
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4775, in _extract_data_from_tsv
    values = list(map(dtype, fields[1:]))
ValueError: could not convert string to float: 'k__test;p__test'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 660, in load_table
    table = parse_biom_table(fp)
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 412, in parse_biom_table
    t = Table.from_tsv(fp, None, None, lambda x: x)
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4631, in from_tsv
    t_md_name) = Table._extract_data_from_tsv(lines, **kwargs)
  File "/usr/lib/python3/dist-packages/biom/table.py", line 4779, in _extract_data_from_tsv
    raise TypeError(msg % (lineno, badidx+1, badval))
TypeError: Invalid value on line 2, column 3, value k__test;p__test

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/biom", line 11, in <module>
    load_entry_point('biom-format==2.1.7', 'console_scripts', 'biom')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/biom/cli/table_converter.py", line 114, in convert
    table = load_table(input_fp)
  File "/usr/lib/python3/dist-packages/biom/parse.py", line 662, in load_table
    raise TypeError("%s does not appear to be a BIOM file!" % f)
TypeError: otutable.txt does not appear to be a BIOM file!
```

</p>
</details>

I used biom-format-tools version 2.1.7+dfsg-2 amd64 installed through APT on Debian GNU/Linux 10 (buster)

### PR DESCRIPTION
**Pull request by wasade**
Fixes #827 

@antgonza, would it be possible for a review?

### CODE REVIEW

File: biom/tests/test_table.py [hunk 1]
```diff
@@ -4123,6 +4164,21 @@ def test_list_sparse_to_sparse(self):
...
+1\t1216\t3500\t6\tBacteria; Firmicutes; Alicyclobacillaceae; Bacilli; La\
+ctobacillales; Lactobacillales; Streptococcaceae; Streptococcus
+7\t1803\t1184\t2\tBacteria; Actinobacteria; Actinobacteridae; Gordoniace\
+ae; Corynebacteriaceae
+3\t1722\t4903\t17\tBacteria; Firmicutes; Alicyclobacillaceae; Bacilli; St\
+aphylococcaceae
+4\t589\t2074\t34\tBacteria; Cyanobacteria; Chloroplasts; vectors
+"""
+extract_tsv_bug = """#OTU ID	s1	s2	taxonomy
+1	123	32\t
```
Comment (wasade) [inline biom/tests/test_table.py]: NOTE: an explicit `\t` is used here as opposed to `^I` as some editors strip dangling whitespace

File: biom/table.py [hunk 2]
```diff
@@ -4724,20 +4732,8 @@ def _extract_data_from_tsv(lines, delim='\t', dtype=float, md_parse=None):
...
-            try:
-                float(last_value)
-            except ValueError:
-                last_column_is_numeric = False
-        else:
-            try:
-                int(last_value)
-            except ValueError:
-                last_column_is_numeric = False
+        last_values = [line.rsplit('\t', 1)[-1].strip() for line in lines]
```
Comment (antgonza) [inline biom/table.py]: should \t be delim?

**Comment by antgonza** [comment]
... and tests are failing ...

File: biom/table.py (hunk 2 above)
Comment (wasade) [reply biom/table.py]: yaya, good catch

**Comment by wasade** [comment]
Ugh. Actual failures from a test module I didn't consider.

**Pull request merged**